tests/integration/test_deletes.py

```diff
@@ -417,3 +452,107 @@ def test_delete_truncate(session_catalog: RestCatalog) -> None:
     assert len(entries) == 1
 
     assert entries[0].status == ManifestEntryStatus.DELETED
+
+
+@pytest.mark.integration
+def test_delete_overwrite_table_with_null(session_catalog: RestCatalog) -> None:
+    arrow_schema = pa.schema([pa.field("ints", pa.int32())])
+    arrow_tbl = pa.Table.from_pylist(
+        [{"ints": 1}, {"ints": 2}, {"ints": None}],
+        schema=arrow_schema,
+    )
+
+    iceberg_schema = Schema(NestedField(1, "ints", IntegerType()))
+
+    tbl_identifier = "default.test_delete_overwrite_with_null"
+
+    try:
+        session_catalog.drop_table(tbl_identifier)
+    except NoSuchTableError:
+        pass
+
+    tbl = session_catalog.create_table(tbl_identifier, iceberg_schema)
+    tbl.append(arrow_tbl)
```

Comment:
[Optional] I like the `_create_table` util we have in other integration tests:
https://github.com/apache/iceberg-python/blob/1ed3abdd1aec480911eeec4f0f46a04efe53dc06/tests/integration/test_add_files.py#L121-L138

It will be great if we can have the same for `test_deletes`. (totally ok to be done in a follow-up PR after the release) 

In the future, we may also want to consider unifying the behavior of `_create_table` across different tests and moving it to `conftest.py`. We can do this in a follow-up PR.

Reply:
sounds good! let's put it in a future pr